src/VisualStudio/CSharp/Impl/ObjectBrowser/CSharpLibraryServiceFactory.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Library;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.ObjectBrowser
+{
+    [ExportLanguageServiceFactory(typeof(ILibraryService), LanguageNames.CSharp), Shared]
+    internal class CSharpLibraryServiceFactory : ILanguageServiceFactory
```

Comment:
This isn't bad, but if you're not doing anything magic in your CreateLanguageService, you can just use ExportLanguageService and skip the factory entirely. But hey, it's written, and half of our code already does this.